source/MaterialXView/ShaderValidators/Glsl/GlslValidator.cpp

```diff
@@ -0,0 +1,1225 @@
+                }
+                else
+                {
+                    pixels[i][j][0] = (GLubyte)(255 * mdist);
+                    pixels[i][j][1] = (GLubyte)(255 * mdist);
+                    pixels[i][j][2] = (GLubyte)(255 * mdist);
+                }
+                pixels[i][j][3] = (GLubyte)255;
+            }
+        }
+
+        glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
+        glGenTextures(1, &_dummyTexture);
+
+        glBindTexture(GL_TEXTURE_2D, _dummyTexture);
+        glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, imageSize, imageSize,
+            0, GL_RGBA, GL_UNSIGNED_BYTE, pixels);
+        // Note: Must do this for default sampling to lookup properly.
+        glGenerateMipmap(GL_TEXTURE_2D);
+        glBindTexture(GL_TEXTURE_2D, UNDEFINED_OPENGL_RESOURCE_ID);
+    }
+}
+
+void GlslValidator::unbindTextures()
+{
+    int textureUnit = 0;
+    GLint maxImageUnits = -1;
+    glGetIntegerv(GL_MAX_COMBINED_TEXTURE_IMAGE_UNITS, &maxImageUnits);
+    for (auto uniform : _uniformList)
+    {
+        GLenum uniformType = uniform.second->_type;
+        GLint uniformLocation = uniform.second->_location;
+        if (uniformLocation >= 0 &&
+            uniformType >= GL_SAMPLER_1D && uniformType <= GL_SAMPLER_CUBE)
+        {
+            if (textureUnit >= maxImageUnits)
+            {
+                break;
+            }
+
+            // Unbind a texture to that unit
+            glActiveTexture(GL_TEXTURE0 + textureUnit);
+            glBindTexture(GL_TEXTURE_2D, UNDEFINED_OPENGL_RESOURCE_ID);
+            checkErrors();
+            textureUnit++;
+        }
+    }
+    glDeleteTextures(1, &_dummyTexture);
+    _dummyTexture = UNDEFINED_OPENGL_RESOURCE_ID;
+
+    checkErrors();
+}
+
+void GlslValidator::bindTextures()
+{
+    if (_programId <= 0)
+    {
+        const std::string errorType("GLSL matrix bind error.");
+        ShaderValidationErrorList errors;
+        errors.push_back("Cannot bind textures without a valid program");
+        throw ExceptionShaderValidationError(errorType, errors);
+    }
+
+    // Pull information from program directly
+    int textureUnit = 0;
+    GLint maxImageUnits = -1;
+    glGetIntegerv(GL_MAX_COMBINED_TEXTURE_IMAGE_UNITS, &maxImageUnits);
+    for (auto uniform : _uniformList)
+    {
+        GLenum uniformType = uniform.second->_type;
+        GLint uniformLocation = uniform.second->_location;
+        if (uniformLocation >= 0 &&
+            uniformType >= GL_SAMPLER_1D && uniformType <= GL_SAMPLER_CUBE)
+        {
+            if (textureUnit >= maxImageUnits)
+            {
+                break;
+            }
+
+            // Map location to a texture unit incrementally
+            glUniform1i(uniformLocation, textureUnit);
+            // Bind a texture to that unit
+            glActiveTexture(GL_TEXTURE0 + textureUnit);
+
+            // Pull information from HwShader as needed
+            bool textureBound = false;
+            if (_hwShader)
```

Comment:
Logic to add to get filename from hwShader. For now always binds a dummy ramp texture.